.github/workflows/release.yml

```diff
@@ -5,141 +5,246 @@ on:
     steps:
-      - uses: actions/checkout@v2
-      - name: Build wasm
-        env:
-          RUST_TOOLCHAIN: nightly-2021-04-22
+      - name: Install dependencies
         run: |
-          .maintain/setup-wasm.sh
+          apt update -y
+          apt install --no-install-recommends -y \
+            git curl ca-certificates
 
-      - name: Upload deploy
-        uses: actions/upload-artifact@v2
+      - uses: actions/download-artifact@v2
         with:
           name: darwinia-artifact
-          path: deploy
+          path: shared
 
-      - uses: 8398a7/action-slack@v3
-        if: failure()
-        with:
-          status: ${{ job.status }}
-          fields: repo,author,eventName,workflow,ref,commit
+      - name: Install subwasm
         env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_INCOMING_WEBHOOK_URL }}
+          # Currently, Same with srtool version v0.9.18
+          # https://github.com/paritytech/srtool/blob/v0.9.18/Dockerfile#L19
+          SUBWASM_VERSION: 0.15.0
+        run: |
+          curl -L https://github.com/chevdor/subwasm/releases/download/v${SUBWASM_VERSION}/subwasm_linux_amd64_v${SUBWASM_VERSION}.deb \
+            --output subwasm.deb && \
+            dpkg -i subwasm.deb && \
+            subwasm --version
 
+      - name: Generate subwasm info for ${{ matrix.runtime }}
+        run: |
+          WASM=shared/${{ matrix.runtime }}/*.compact.wasm
+          Z_WASM=shared/${{ matrix.runtime }}/*.compact.compressed.wasm
+
+          INFO_WASM=$(subwasm -j info $WASM)
+          INFO_Z_WASM=$(subwasm -j info $Z_WASM)
+
+          echo "$INFO_WASM" > shared/${{ matrix.runtime }}/${{ matrix.runtime }}.compact.json
+          echo "$INFO_Z_WASM" > shared/${{ matrix.runtime }}/${{ matrix.runtime }}.compact.compressed.json
+
+      - name: Upload shared
+        uses: actions/upload-artifact@v2
+        with:
+          name: darwinia-artifact
+          path: shared
 
   build-image:
     name: Build image
     runs-on: ubuntu-latest
-    needs: [build-binary]
+    needs: [build-package]
     steps:
       - uses: actions/checkout@v2
 
+      - uses: actions/download-artifact@v2
+        with:
+          name: darwinia-artifact
+          path: shared
+
       - name: Docker login
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.QUAY_IO_BOT_USERNAME }}
           password: ${{ secrets.QUAY_IO_BOT_PASSWORD }}
           registry: quay.io
 
-      - name: Download deploy
-        uses: actions/download-artifact@v2
+      - uses: olegtarasov/get-tag@v2.1
+        id: tag-name
+      - uses: benjlevesque/short-sha@v1.2
+        id: short-sha
         with:
-          name: darwinia-artifact
-          path: deploy
+          length: 7
 
-      - name: Setup image
+      - name: Build and publish docker image
         env:
-          REGISTRY_HOST: quay.io
+          SHA: ${{ env.SHA }}
         run: |
-          .maintain/setup-image.sh
+          chmod +x shared/darwinia
```

Comment:
The binary will automatically get the executive permission after `cargo build`. Is this required?